File: Source/3 Data mining/Implementation - Jan/Java/src/itu/datamining/week4/ItemSet.java
package itu.datamining.week4;

/***
 * The ItemSet class is used to store information concerning a single transaction.
 * Should not need any changes.
 * @author andershh, jang
 *
 */
public class ItemSet {
	
	/***
	 * The PRIMES array is internally in the ItemSet-class' hashCode method
	 */
	private static final int[] PRIMES = { 2, 3, 5, 7, 11, 13, 17, 23, 27, 31, 37 };
    final int[] set;

    /***
     * Creates a new instance of the ItemSet class.
     * @param set Transaction content
     */
    public ItemSet( int[] set ) {
        this.set = set;
    }

    @Override
    /**
     * hashCode functioned used internally in Hashtable
     */
    public int hashCode() {
        int code = 0;
        for (int i = 0; i < set.length; i++) {
            code += set[i] * PRIMES[i];
        }
        return code;
    }

    
    @Override
    /**
     * Used to determine whether two ItemSet objects are equal
     */
    public boolean equals( Object o ) {
        if (!(o instanceof ItemSet)) {
            return false;
        }
        ItemSet other = (ItemSet) o;
        if (other.set.length != this.set.length) {
            return false;
        }
        for (int i = 0; i < set.length; i++) {
            if (set[i] != other.set[i]) {
                return false;
            }
        }
        return true;
    }
}

File: Source/3 Data mining/Implementation - Jan/Dotnet/Itu.Jve.Datamining/DataminingConsole/Processes/DataMiningSpring2016/PatternDiscovery/Apriori/ItemSEt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataminingConsole.Processes.DataMiningSpring2016.PatternDiscovery.Apriori
{
    public class ItemSet : IComparable
    {
        /***
            * The PRIMES array is internally in the ItemSet-class' hashCode method
        */
        private static int[] PRIMES = { 2, 3, 5, 7, 11, 13, 17, 23, 27, 31, 37 };
        public int[] Set { get; set; }

        /**
        * hashCode functioned used internally in Hashtable
        */
        public override int GetHashCode()
        {
            int code = 0;
            for (int i = 0; i < Set.Length; i++)
            {
                code += Set[i] * PRIMES[i];
            }
            return code;
        }

        /**
        * Used to determine whether two ItemSet objects are equal
        */
        public override bool Equals(object obj)
        {
            if (obj == null) return false;
            var other = obj as ItemSet;

            if (other != null)
            {
                if (other.Set.Length != this.Set.Length)
                {
                    return false;
                }
                for (int i = 0; i < Set.Length; i++)
                {
                    if (Set[i] != other.Set[i])
                    {
                        return false;
                    }
                }
                return true;
            }
            else
                throw new ArgumentException("Object is not an AgeAttribute");
        }

        public int CompareTo(object obj)
        {
            throw new NotImplementedException();
        }
    }
}
